Ginger/Ginger/Drivers/DriversWindows/MobileDriverWindow.xaml.cs

```diff
@@ -1054,9 +1529,96 @@ private void PerformScreenSwipe(eSwipeSide swipeSide, double impact = 1)
+                    docPage.Width = 800;
+                    docPage.Height = 800;
+                    GingerCore.General.LoadGenericWindow(ref genWin, App.MainWindow, eWindowShowStyle.Free, title, docPage);
+
+                });
+
+                System.IO.File.Delete(tempFilePath);
+                return;
+            }
+        }
+
+
 
+        private void xAgentBtn_Click(object sender, RoutedEventArgs e)
+        {
+            Application.Current.Dispatcher.Invoke(() =>
+            {
+                AgentEditPage agentEditPage = new AgentEditPage(mAgent);
+                agentEditPage.Width = 800;
+                agentEditPage.Height = 800;
+                ((Grid)((ScrollViewer)agentEditPage.Content).Content).IsEnabled = false;
+                GingerCore.General.LoadGenericWindow(ref genWin, App.MainWindow, eWindowShowStyle.Free, mAgent.Name, agentEditPage);
+            });
+        }
+
+        private void xRamUsageBtn_Click(object sender, RoutedEventArgs e)
+        {
+            OpenPopUpWindow(DictionaryToString(mDeviceMemoryInfo).Replace(", ", Environment.NewLine), "Full device's Memory Information");
+        }
+
+        private void xNetworksBtn_Click(object sender, RoutedEventArgs e)
+        {
+            OpenPopUpWindow(mDeviceNetworkInfo, "Full device's Network Information");
+        }
+
+
+
+        //private void xAgentValueLbl_TargetUpdated(object sender, System.Windows.Data.DataTransferEventArgs e)
+        //{
+        //    string labelContent = xAgentValueLbl.Content.ToString();
+        //    if (labelContent.Length > 35)
+        //    {
+        //        xAgentValueLbl.Content = labelContent.Substring(0, 35) + "...";
+        //    }
+        //}
+
+        private void xExtraInfoBtn_ExtraInfo_Click(object sender, RoutedEventArgs e)
+        {
+            DeviceInfo deviceInfo = (DeviceInfo)((ucButton)sender).DataContext;
+            switch (deviceInfo.DetailName)
+            {
+                case "App RAM Usage:":
+                    OpenPopUpWindow(DictionaryToString(mDeviceMemoryInfo).Replace(", ", Environment.NewLine), "Full device's Memory Information");
+                    break;
+                case "Networks:":
+                    OpenPopUpWindow(mDeviceNetworkInfo, "Full device's Network Information");
+                    break;
+                case "App CPU Usage:":
+                    OpenPopUpWindow(DictionaryToString(mDeviceCPUInfo).Replace(", ", Environment.NewLine), "Full device's CPU Information");
+                    break;
+                case "Ginger Agent:":
+                    Application.Current.Dispatcher.Invoke(() =>
+                    {
+                        AgentEditPage agentEditPage = new AgentEditPage(mAgent);
+                        agentEditPage.Width = 1000;
+                        agentEditPage.Height = 1000;
+                        ((TextBox)agentEditPage.FindName("xAgentNameTextBox")).IsReadOnly = true;
```

Comment:
that is not pretty way for showing in read only- need to have flag on the page to load it as REadOnly and then to handle it inside the page and not externally